### ISSUE TITLE
Delete Functionality

### ISSUE THREAD

**Issue by MichaelJCompton** [MichaelJCompton/ConceptDiagrams#1]
At present delete only works for a single item and only 'bottom-up' : e.g. need to delete everything in a diagram before can delete the diagram and delete incoming/outgoing arrows before we can delete a curve.  

Might be nice to have delete work with these groups.  E.g. as you hover over a curve it highlights, if in delete mode it would highlight the curve and anything necessary to also delete the curve, similarly for a boundary rectangle it would highlight the rectangle and anything in it as well as curves in and out.

The code (currently DiagramPresenter + commands) would then work out a delete order and batch out a multi-delete that could be done and undone in one step.

Similarly, for groups if allowed in the interface.

**Comment by MichaelJCompton** [comment]
Should have noted that a multi-command would just be a list of commands and would cycle through executing/unexecuting them all.
